src/components/cylc/tree/cylc-tree.js

```diff
@@ -16,7 +16,7 @@
  */
 import { computePercentProgress } from '@/components/cylc'
 import { extractGroupState } from '@/utils/tasks'
-import { merge } from 'lodash'
+import { merge, sortedIndex } from 'lodash'
```

Comment:
A utility function from Lodash that does binary-search to locate the next index to add an element to a sorted array. This is used for every `add....` function of the tree.

The cycle points are the only ones sorted in reverse order (the newest first), so we adjust the index for that case.